framework/test/unit/specs/modules/token/token_module.spec.ts

```diff
@@ -67,17 +72,121 @@ describe('token module', () => {
 				balance: BigInt('1000000000000000'),
 			},
 		});
-		stateStore = new StateStoreMock([senderAccount, recipientAccount]);
-		stateStore.account = {
-			...stateStore.account,
-			get: jest.fn().mockResolvedValue(senderAccount),
-			getOrDefault: jest.fn().mockResolvedValue(senderAccount),
+		genesisBlock = {
+			header: {
+				asset: {
+					accounts: [senderAccount],
+				},
+			},
 		};
+		stateStore = new StateStoreMock([senderAccount, recipientAccount]);
+		jest.spyOn(stateStore.account, 'getOrDefault').mockResolvedValue(senderAccount);
+		jest.spyOn(stateStore.account, 'get').mockResolvedValue(senderAccount);
+		jest.spyOn(stateStore.account, 'set');
+		jest.spyOn(stateStore.chain, 'get');
+		jest.spyOn(stateStore.chain, 'set');
+
 		reducerHandler = {};
 	});
 
+	describe('#reducers.credit', () => {
+		it('should throw error if address is not a buffer', async () => {
+			return expect(
+				tokenModule.reducers.credit({ address: 'address', amount: BigInt('1000') }, stateStore),
+			).rejects.toStrictEqual(new Error('Address must be a buffer'));
+		});
+
+		it('should throw error if amount is not a bigint', async () => {
+			return expect(
+				tokenModule.reducers.credit({ address: senderAccount.address, amount: '1000' }, stateStore),
+			).rejects.toStrictEqual(new Error('Amount must be a bigint'));
+		});
+
+		it('should throw error if account does not have sufficient balance', async () => {
+			senderAccount.token.balance = BigInt(0);
+
+			return expect(
+				tokenModule.reducers.credit(
+					{ address: senderAccount.address, amount: BigInt('1000') },
+					stateStore,
+				),
+			).resolves.toBeUndefined();
```

Comment:
description and assertion is not matching. I think it should be `.rejects.toThrow`?